Packs/FalconX/Integrations/FalconX/FalconX.py

```diff
@@ -0,0 +1,415 @@
+        headers = {
+            'Authorization': 'bearer ' + self.token,
+        }
+
+        return self.http_request("Get", url_suffix, headers=headers, params=params)
+
+    def _generate_token(self) -> str:
+        """Generate an Access token
+
+        Returns:
+            valid token
+        """
+
+        body = {
+            'client_id': self._username,
+            'client_secret': self._password
+        }
+
+        BYTE_CREDS = f'{self._username}:{self._password}'.encode('utf-8')
+
+        HEADERS = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Authorization': f'Basic {base64.b64encode(BYTE_CREDS).decode()}'
+        }
+
+        headers = {
+            'Authorization': HEADERS['Authorization']
+        }
+        token_res = self.http_request('POST', '/oauth2/token', data=body, headers=headers)
+        # print(token_res.text)
+        return token_res.get('access_token')
+
+
+def _create_short_context_falconx(resources: dict):
+    sandbox = resources.get("sandbox")[0]
+    resource_output = {
+        'id': resources.get("id"),
+        'state': resources.get("state"),
+        'created_timestamp': resources.get("created_timestamp"),
+        'sha256': sandbox.get("sha256"),
+        'environment_id': sandbox.get("environment_id")
+    }
+    return {f'csfalconx.resource(val.resource === obj.resource)': resource_output}
+
+
+def _create_full_context_falconx(resources: dict):
+    sandbox = resources.get("sandbox")[0]
+    resource_output = {
+        'id': resources.get("id"),
+        'verdict': resources.get("verdict"),
+        'created_timestamp': resources.get("created_timestamp"),
+        'environment_id': sandbox.get("environment_id"),
+        'environment_description': sandbox.get("environment_description"),
+        'sandbox_threat_score': sandbox.get("threat_score"),
+        'sandbox_submit_url': sandbox.get("submit_url"),
+        'submission_type': sandbox.get("submission_type"),
+        'sandbox_filetyp': sandbox.get("filetyp"),  # find
+        'sandbox_filesize': sandbox.get("filesize"),  # find
+        'sandbox_sha256': sandbox.get("sha256"),
+        'ioc_strict_csv': resources.get("ioc_report_strict_csv_artifact_id"),
+        'ioc_broad_csv': resources.get("ioc_report_broad_csv_artifact_id"),
+        'ioc_strict_jason': resources.get("ioc_report_strict_json_artifact_id"),
+        'ioc_broad_jason': resources.get("ioc_report_broad_json_artifact_id"),
+        'ioc_strict_stix': resources.get("ioc_report_strict_stix_artifact_id"),
+        'ioc_broad_stix': resources.get("ioc_report_broad_stix_artifact_id"),
+        'ioc_strict_maec': resources.get("ioc_report_strict_maec_artifact_id"),
+        'ioc_broad_maec': resources.get("ioc_report_broad_maec_artifact_id"),
+    }
+    return {f'csfalconx.resource(val.resource === obj.resource)': resource_output}
+
+
+def test_module(client):
```

Comment:
add hints/types.
we shouldalso  run the `cs-fx-check-quota` and if quota is not greater than 0, print an informative error message.